pkg/provisioner/ironic/ironic.go

```diff
@@ -1152,6 +1152,53 @@ func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings me
 	return settings, schema, nil
 }
 
+// GetFirmwareComponents gets all available firmware components for a node and return a list.
+func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareComponentStatus, error) {
+	ironicNode, err := p.getNode()
+	if err != nil {
+		return nil, fmt.Errorf("could not get node to retrieve firmware components: %w", err)
+	}
+
+	if !p.availableFeatures.HasFirmwareUpdates() {
+		return nil, fmt.Errorf("current ironic version does not support firmware updates")
+	}
+	// Get the components from Ironic via Gophercloud
+	componentList, componentListErr := nodes.ListFirmware(p.ctx, p.client, ironicNode.UUID).Extract()
+
+	if componentListErr != nil {
+		bmcAccess, _ := p.bmcAccess()
+		if bmcAccess.FirmwareInterface() == "no-firmware" {
+			return nil, fmt.Errorf("node %s is using firmware interface %s: %w", ironicNode.UUID, bmcAccess.FirmwareInterface(), componentListErr)
```

Comment:
For a follow-up: we should not include node UUIDs or refer to nodes or interfaces in user-visible messages since users don't have access to Ironic. Here we should talk about the driver not supporting firmware updates.

Reply:
https://github.com/metal3-io/baremetal-operator/pull/1655 will take care of this and the other parts we discussed